### ISSUE TITLE
Unreal Tournament (1999) CD install fails with FileNotFoundError

### ISSUE THREAD

**Issue by its-thomas** [lutris/lutris#4600]
Hi there,

I am running a Debian 11 (stable) system, having installed Lutris through [flatpak](https://flathub.org/apps/details/net.lutris.Lutris) via the regular "flatpak install flathub net.lutris.Lutris" command with multiple dependencies. So far, so good.

Now, I am trying to install Unreal Tournament (1999 ) with an original store-bought CD via the following route in Lutris:

1. Choosing option "Search the Lutris website for installers"
2. Choosing option "Wine | Windows CD + OldUnreal Patches"

Everything progresses without issues, I even get the Unreal tournament message "Installation Completed Succesfully!"

But then – as far as I can tell, when the Lutris installer tries to install the patches (which have successfully been downloaded during installation!) – I get an "FileNotFoundError" error. And that's it. Game over – so to speak.

Any help would be greatly appreciated!

Thanks
Thomas

**Comment by danieljohnson2** [comment]
We'll need a bit more than that to help.

Run lutris from the command line in debug mode, with "lutris -d". This will emit a lot more log output.

Try to install Unreal Tournament like that, and then give us the log output. There may be clue in there somewhere!

**Comment by its-thomas** [comment]
Thanks for getting back so quickly, @danieljohnson2!

I just ran "flatpak run net.lutris.Lutris -d" and pasted the whole output [here](https://controlc.com/bab40e3e).

**Comment by its-thomas** [comment]
Thanks so much for your efforts, @danieljohnson2!

As I have stated before, I am using the flatpak version of Lutris. So I assume, downloading the latest master and running the script "bin/lutris" within will not work for me, I assume?

**Comment by danieljohnson2** [comment]
I'd overlooked that.

It *might* work anyway. However, even if it does it won't put the game into the flatpak version's personal game archive- it will have its own. Though you could place the game in the same ~/Games/unreal-tournament install directory anyway.

It's a bit messy; non-flatpak Lutris will download stuff into ~/.local/share/lutris, ~/.cache/lutris and ~/.config/lutris for you to clean up later if you care about that sort of thing. And maybe it will fail since you won't have all the dependencies flatpak would have provided.

I'd provide a flatpak test version if I knew how to do that; maybe the only thing we can do is wait for the release, and I think that should wait for #4576. I think it'll be fixed. Probably.

**Comment by danieljohnson2** [comment]
Not sure. If you can find the file, and flatpak itself does not object to it, that might just work.

I think it might be somewhere like this:
~/.local/share/flatpak/app/net.lutris.Lutris/x86_64/beta/e91467190d67425d4c5ffd48edc3425ec1bd1a38223f4dcf31019a55354fb5d0/files/lib/python3.10/site-packages/lutris/util/extract.py

But maybe your random bunch of numbers will be different from mine.

**Comment by its-thomas** [comment]
Unfortunately, I was shown the same FileNotFoundError-error (talking about the superficial user experience level here!) at the same step of the installation as before. See debug [protocol](https://controlc.com/6cf8bb71) for details.

Still, the error message at the end of the protocol seem to be identical ... could I have replaced the wrong instance of extract.py?

FYI: I have used the command "find / -name extract.py 2>/dev/null" via the terminal to search my whole system. And the only instance, where an extract.py file had "lutris" or "Lutris" in it's path was the before mentioned:

/var/lib/flatpak/app/net.lutris.Lutris/x86_64/stable/46f4d8bdde456e6f056b02cb78221837ddbee9f6db8a8b1efe6842299d424ca0/files/lib/python3.9/site-packages/lutris/util/extract.py

**Comment by its-thomas** [comment]
Stop the presses! It has worked!

What did I do?
Well, the same thing I did about 6 hours ago: I copy & pasted the RAW content of the extract.py from github in a local file and replaced it with the current file on my system. But 6 hours ago, the content of the extract.py on github was about 0,7kB less than it is now (I have compared both files).

Could it be that 6 hours ago the changes were not yet available on github when selecting the RAW code view? Or did I do some stupid mistake?!
